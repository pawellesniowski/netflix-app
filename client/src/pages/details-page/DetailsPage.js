import React from 'react';
import { connect } from "react-redux";
import MovieDetails from "../../components/MovieDetails/MovieDetails";
import MoviesGrid from "../../components/MoviesGrid/MoviesGrid";
import Footer from "../../components/Footer/Footer";
import SortBar from "../../components/SortBar/SortBar";
import Navigation from "../../components/Navigation/Navigation";
import { fetchFilm } from './detailsPageActions';
import { withRouter } from 'react-router-dom'

class DetailsPage extends React.Component {
    componentDidMount() {
        this.props.fetchFilm(this.props.match.params.id);
    }

    componentDidUpdate(prevProps) {
        const filmId = this.props.match.params.id;
        if (prevProps.match.params.id !== filmId) {
            this.props.fetchFilm(filmId);
        }
    }

    render() {
        return (
            <React.Fragment>
                <Navigation/>
                <MovieDetails selectedMovie={this.props.film}/>
                <SortBar moviesNumber={this.props.suggestedFilms.length}/>
                <MoviesGrid movies={this.props.suggestedFilms}/>
                <Footer/>
            </React.Fragment>
        )
    }
}

function mapStateToProps(state) {
    return {
        film: state.detailsPage.film,
        suggestedFilms: state.detailsPage.suggestedFilms,
    }
}

const mapDispatchToProps = {
    fetchFilm,
};

export default withRouter(connect(mapStateToProps, mapDispatchToProps)(DetailsPage));
