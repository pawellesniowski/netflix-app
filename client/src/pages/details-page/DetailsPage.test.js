import React from 'react';
import {shallow} from "enzyme/build";
import DetailsPage from "./DetailsPage";

describe('DetailswPage snapshot', () => {
    it("Renders DetailsPage Component", () => {
        const component = shallow(<DetailsPage/>);
        expect(component).toMatchSnapshot();
    });
});
