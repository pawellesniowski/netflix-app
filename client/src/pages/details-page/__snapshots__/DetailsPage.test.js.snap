// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DetailswPage snapshot Renders DetailsPage Component 1`] = `ShallowWrapper {}`;
